### ISSUE TITLE
`useLocalizedStringFormatter` is not documented

### ISSUE THREAD

**Issue by Itherael** [adobe/react-spectrum#3735]
# 🙋 Documentation Request

`useMessageFormatter` is hidden from the sidebar, and the new hook `useLocalizedStringFormatter` is not documented.

There is no official documentation for translating strings in [React Aria documentation](https://react-spectrum.adobe.com/react-aria).

React Aria has documentation for handling dates and numbers but not strings.

**Comment by LFDanLu** [comment]
Thank you for filing, we've added this to our backlog and will try to get to it when time permits!

**Comment by ritz078** [comment]
It also looks like this there is error in the compiled code.

https://github.com/adobe/react-spectrum/blob/6c61f5b1db5f52ebc654bf754a15594bb0662f6f/packages/%40internationalized/string/src/LocalizedStringFormatter.ts#L37

Whenever I use this hook, it throws an error because `this` here is undefined. Everything else works fine.

**Comment by pidupuis** [comment]
Any progress on this?

**Comment by yihuiliao** [comment]
We plan to update our hooks in our new documentation so unlikely we'd update it until we our new doc site goes up

**Comment by achievecreative** [comment]
When will the new documentation site go live?

**Comment by snowystinger** [comment]
We are working on it as fast as we can, it's our top priority right now, thank you for your patience.

**Comment by devongovett** [comment]
I don't think we intend to document this in the foreseeable future. I would recommend using an industry standard library such [react-intl](https://formatjs.github.io/docs/react-intl/) for your translation needs. `useLocalizedStringFormatter` is used internally for some very specific constraints that we have, but wasn't designed to be used externally. It requires a specific build tool to compile the strings for example, which we don't intend to port to all bundlers. A fully supported external library would be a much better recommendation.
